### ISSUE TITLE
Travis should run the demo notebooks as system tests, and deploy the rendered versions as supporting documentation

### ISSUE THREAD

**Issue by drphilmarshall** [LSSTDESC/SLRealizer#50]
Here's the to-do-list:

* [ ] Make travis run our demo notebooks, following the example in [SLTimer](https://github.com/LSSTDESC/SLTimer/blob/master/.travis.yml)
* [ ] Have travis deploy the rendered versions to an orphan `rendered` branch, in the same way that [beavis-ci]() does. 
* [ ] Make the API docs point at the rendered notebooks. 

The problem is that the notebooks exist in the `issue/17/desc-note` branch, while the API docs are also not yet merged. I suggest we do the first two items in a sub-branch off of `issue/17/desc-note`, while getting the API docs merged into `master` in parallel. Then we can pull in the API docs to the Note branch too.

**Comment by jiwoncpark** [comment]
Correction: the demo notebooks are in all the branches, not just the `issue/17/desc-note` branch, so we will branch off of `master` (`issue/50/notebooks-ci') to fix this issue. The point about getting `issue/43/api-docs` merged into `master` in parallel stands. Once both branches are merged, we can have the API docs point at the rendered notebooks.

Currently, the demo notebooks are hiding in `slrealizer/demo` so we'll make them more visible by bring them to `doc` while flattening `doc/notebooks` and `doc/tutorials`.
